Framework/Algorithms/test/JoinRunsTest.h

```diff
@@ -0,0 +1,250 @@
+
+  void setUp() override {
+    MatrixWorkspace_sptr ws1 = create2DWorkspace123(5, 3); // 3 points
+    MatrixWorkspace_sptr ws2 = create2DWorkspace154(5, 2); // 2 points
+    MatrixWorkspace_sptr ws3 = create2DWorkspace123(5, 1); // 1 point
+    MatrixWorkspace_sptr ws4 = create2DWorkspace154(5, 1); // 1 point
+
+    ws1->getAxis(0)->setUnit("TOF");
+    ws2->getAxis(0)->setUnit("TOF");
+    ws3->getAxis(0)->setUnit("TOF");
+    ws4->getAxis(0)->setUnit("TOF");
+
+    storeWS("ws1", ws1);
+    storeWS("ws2", ws2);
+    storeWS("ws3", ws3);
+    storeWS("ws4", ws4);
+
+    m_testWS = {"ws1", "ws2", "ws3", "ws4"};
+  }
+
+  void tearDown() override {
+    removeWS("ws1");
+    removeWS("ws2");
+    removeWS("ws3");
+    removeWS("ws4");
+    m_testWS.clear();
+  }
+
+  void testInit() {
+    TS_ASSERT_THROWS_NOTHING(m_testee.initialize());
+    TS_ASSERT(m_testee.isInitialized());
+  }
+
+  void testHappyCase() {
+    m_testee.setProperty("InputWorkspaces", m_testWS);
+    m_testee.setProperty("OutputWorkspace", "out");
+    TS_ASSERT_THROWS_NOTHING(m_testee.execute());
+    TS_ASSERT(m_testee.isExecuted());
+
+    MatrixWorkspace_sptr out =
+        AnalysisDataService::Instance().retrieveWS<MatrixWorkspace>("out");
+
+    TS_ASSERT(out);
+    TS_ASSERT_EQUALS(out->getNumberHistograms(), 5);
+    TS_ASSERT_EQUALS(out->blocksize(), 7);
+    TS_ASSERT(!out->isHistogramData());
+    TS_ASSERT_EQUALS(out->getAxis(0)->unit()->unitID(), "TOF");
+
+    std::vector<double> spectrum = out->y(0).rawData();
+    std::vector<double> error = out->e(0).rawData();
+    std::vector<double> xaxis = out->x(0).rawData();
+
+    TS_ASSERT_EQUALS(spectrum[0], 2.);
+    TS_ASSERT_EQUALS(spectrum[1], 2.);
+    TS_ASSERT_EQUALS(spectrum[2], 2.);
+    TS_ASSERT_EQUALS(spectrum[3], 5.);
+    TS_ASSERT_EQUALS(spectrum[4], 5.);
+    TS_ASSERT_EQUALS(spectrum[5], 2.);
+    TS_ASSERT_EQUALS(spectrum[6], 5.);
+
+    TS_ASSERT_EQUALS(error[0], 3.);
+    TS_ASSERT_EQUALS(error[1], 3.);
+    TS_ASSERT_EQUALS(error[2], 3.);
+    TS_ASSERT_EQUALS(error[3], 4.);
+    TS_ASSERT_EQUALS(error[4], 4.);
+    TS_ASSERT_EQUALS(error[5], 3.);
+    TS_ASSERT_EQUALS(error[6], 4.);
+
+    TS_ASSERT_EQUALS(xaxis[0], 1.);
+    TS_ASSERT_EQUALS(xaxis[1], 2.);
+    TS_ASSERT_EQUALS(xaxis[2], 3.);
+    TS_ASSERT_EQUALS(xaxis[3], 1.);
+    TS_ASSERT_EQUALS(xaxis[4], 2.);
+    TS_ASSERT_EQUALS(xaxis[5], 1.);
+    TS_ASSERT_EQUALS(xaxis[6], 1.);
+  }
+
+  void testFailWithNumLog() {
+    AddSampleLog logAdder;
+    logAdder.initialize();
+    logAdder.setProperty("LogName", "TestLog");
+    logAdder.setProperty("LogType", "Number");
+
+    logAdder.setProperty("Workspace", "ws1");
+    logAdder.setProperty("LogText", "0.7");
+    logAdder.execute();
+
+    m_testee.setProperty("InputWorkspaces", m_testWS);
+
+    m_testee.setProperty("SampleLogAsXAxis", "TestNumLog");
+
+    // blocksize must be one in case of scalar log, fail
+    TS_ASSERT_THROWS(m_testee.execute(), std::runtime_error);
```

Comment:
It is nice to assert on the error message too here, to ensure that a good error message is generated and things are failing for the expected reason.

```cpp
TS_ASSERT_THROWS_EQUALS(
    builder.setHistogram(Histogram(x, y)), const std::logic_error &e,
    std::string(e.what()),
    "Histogram supplied does not have the correct size.")
```